src/main/java/pl/psnc/dei/service/PersistableExceptionService.java

```diff
@@ -0,0 +1,127 @@
+package pl.psnc.dei.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.psnc.dei.exception.NotFoundException;
+import pl.psnc.dei.iiif.ConversionException;
+import pl.psnc.dei.iiif.ConversionImpossibleException;
+import pl.psnc.dei.model.DAO.PersistableExceptionRepository;
+import pl.psnc.dei.model.PersistableException;
+import pl.psnc.dei.model.Record;
+import pl.psnc.dei.model.conversion.Context;
+import pl.psnc.dei.model.exception.TranscriptionPlatformException;
+import pl.psnc.dei.service.context.ContextMediator;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class PersistableExceptionService {
+
+    @Autowired
+    private PersistableExceptionRepository persistableExceptionRepository;
+
+    @Autowired
+    private ContextMediator contextMediator;
+
+    public <T extends  Exception> T findByContextAndExceptionClass(Context context, Class<T> exceptionClass) {
+        PersistableException.ExceptionType exceptionType = this.convertExceptionClassToExceptionType(exceptionClass);
+        Optional<PersistableException> optionalExceptions = this.persistableExceptionRepository.findByContextAndType(context, exceptionType);
```

Comment:
rename variable as plural suggests it's a collection of exceptions

Reply:
renamed